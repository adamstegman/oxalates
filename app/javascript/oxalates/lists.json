[
  {
    "id": "all",
    "name": "All",
    "bottom_threshold": null,
    "top_threshold": null
  },
  {
    "id": 1,
    "name": "Very High",
    "bottom_threshold": 50,
    "top_threshold": null
  },
  {
    "id": 2,
    "name": "High",
    "bottom_threshold": 15,
    "top_threshold": 50
  },
  {
    "id": 3,
    "name": "Moderate",
    "bottom_threshold": 5,
    "top_threshold": 15
  },
  {
    "id": 4,
    "name": "Low",
    "bottom_threshold": null,
    "top_threshold": 5
  }
]
