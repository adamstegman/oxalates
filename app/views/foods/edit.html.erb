<%= form_for @food do |f| %>
  <%= f.label :name %>
  <%= f.text_field :name %>
  <%= f.label :list %>
  <%= f.select :list, options_from_collection_for_select(all_lists, :id, :name, @food.list_id) %>

  <%= f.submit "Save" %>
<% end %>
