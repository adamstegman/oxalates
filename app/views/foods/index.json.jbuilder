json.foods(@foods) do |food|
  json.(food, :id, :name, :oxalate_mg, :serving)
end
