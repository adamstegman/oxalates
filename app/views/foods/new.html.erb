<%= form_for @food, html: {class: 'edit-food'} do |f| %>
  <p><%= f.label :name, "New food name" %></p>
  <p><%= f.text_field :name %></p>
  <p><%= f.label :oxalate_mg, "Oxalates per serving (mg)" %></p>
  <p><%= f.number_field :oxalate_mg, step: "any" %></p>
  <p><%= f.label :serving %></p>
  <p><%= f.text_field :serving %></p>

  <p><%= f.submit "Add" %></p>
<% end %>
