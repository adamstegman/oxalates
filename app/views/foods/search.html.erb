<h1>Results</h1>

<ul>
	<% @results.each do |r| %>
		<li><%= r.name + ": " + r.list.name %></li> 
	<% end %>
</ul>