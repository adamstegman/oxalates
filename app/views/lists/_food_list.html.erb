<ul class="food-list">
<% foods.each do |food| %>
  <li class="food-list-item <%= list_class_name(food.list) %>"><%= food.name %></li>
<% end %>
</ul>
