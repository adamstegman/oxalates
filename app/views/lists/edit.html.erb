<%= form_tag list_path(@list.id), method: :put do %>
  <ul class="food-list">
  <% @list.foods.each do |food| %>
    <li>
      <%= link_to food.name, edit_food_path(food), class: "edit-food-list-item #{list_class_name(food.list)}" %>
      <%= hidden_field_tag 'list[foods][][id]', food.id %>
      <%= label_tag 'list[foods][][name]', "Name", class: 'food-list-item-name' %>
      <%= text_field_tag 'list[foods][][name]', food.name, class: 'food-list-item-name' %>
    </li>
  <% end %>
  </ul>
<% end %>
