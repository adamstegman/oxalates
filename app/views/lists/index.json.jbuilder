json.lists(@lists) do |list|
  json.(list, :id, :name, :bottom_threshold, :top_threshold)
end
