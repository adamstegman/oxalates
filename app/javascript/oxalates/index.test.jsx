import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';

import { Oxalates } from './index';
import foods from './__mocks__/foods.json';
import lists from './__mocks__/lists.json';

Enzyme.configure({ adapter: new Adapter() });

const mockStore = configureStore();
const state = {
  listMenu: {
    lists,
    activeListId: lists[0].id,
  },
  foodList: {
    foods,
    error: null,
  },
};

test('Index renders the home screen', () => {
  const store = mockStore(state);
  const component = renderer.create(
    <Provider store={store}>
      <Oxalates />
    </Provider>,
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
