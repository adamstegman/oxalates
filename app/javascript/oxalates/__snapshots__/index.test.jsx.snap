// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index renders the home screen 1`] = `
<div>
  <div
    className="body"
  >
    <div>
      <h1
        className="title all"
      >
        All Foods
      </h1>
    </div>
    <div
      className="content"
    />
  </div>
  <ul
    className="list-menu"
  >
    <li
      className="list-menu-item active"
    >
      <button
        className="list-menu-item-content no-outline"
        onClick={[Function]}
        type="button"
      >
        <div
          className="list-menu-item-icon"
        >
          <svg
            height="25"
            width="21"
          >
            <rect
              fill="none"
              height="25"
              stroke="black"
              strokeWidth="3"
              width="21"
              x="0"
              y="0"
            />
            <line
              stroke="black"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="5"
              y2="5"
            />
            <line
              stroke="#e00"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="10"
              y2="10"
            />
            <line
              stroke="#fa0"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="15"
              y2="15"
            />
            <line
              stroke="green"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="20"
              y2="20"
            />
          </svg>
        </div>
        <p
          className="list-menu-item-text"
        >
          All
        </p>
      </button>
    </li>
    <li
      className="list-menu-item "
    >
      <button
        className="list-menu-item-content no-outline"
        onClick={[Function]}
        type="button"
      >
        <div
          className="list-menu-item-icon"
        >
          <svg
            height="25"
            width="21"
          >
            <rect
              fill="none"
              height="25"
              stroke="black"
              strokeWidth="3"
              width="21"
              x="0"
              y="0"
            />
            <line
              stroke="black"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="5"
              y2="5"
            />
            <line
              stroke="black"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="10"
              y2="10"
            />
            <line
              stroke="black"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="15"
              y2="15"
            />
            <line
              stroke="black"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="20"
              y2="20"
            />
          </svg>
        </div>
        <p
          className="list-menu-item-text"
        >
          Very High
        </p>
      </button>
    </li>
    <li
      className="list-menu-item "
    >
      <button
        className="list-menu-item-content no-outline"
        onClick={[Function]}
        type="button"
      >
        <div
          className="list-menu-item-icon"
        >
          <svg
            height="25"
            width="21"
          >
            <rect
              fill="none"
              height="25"
              stroke="black"
              strokeWidth="3"
              width="21"
              x="0"
              y="0"
            />
            <line
              stroke="#e00"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="5"
              y2="5"
            />
            <line
              stroke="#e00"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="10"
              y2="10"
            />
            <line
              stroke="#e00"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="15"
              y2="15"
            />
            <line
              stroke="#e00"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="20"
              y2="20"
            />
          </svg>
        </div>
        <p
          className="list-menu-item-text"
        >
          High
        </p>
      </button>
    </li>
    <li
      className="list-menu-item "
    >
      <button
        className="list-menu-item-content no-outline"
        onClick={[Function]}
        type="button"
      >
        <div
          className="list-menu-item-icon"
        >
          <svg
            height="25"
            width="21"
          >
            <rect
              fill="none"
              height="25"
              stroke="black"
              strokeWidth="3"
              width="21"
              x="0"
              y="0"
            />
            <line
              stroke="#fa0"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="5"
              y2="5"
            />
            <line
              stroke="#fa0"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="10"
              y2="10"
            />
            <line
              stroke="#fa0"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="15"
              y2="15"
            />
            <line
              stroke="#fa0"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="20"
              y2="20"
            />
          </svg>
        </div>
        <p
          className="list-menu-item-text"
        >
          Moderate
        </p>
      </button>
    </li>
    <li
      className="list-menu-item "
    >
      <button
        className="list-menu-item-content no-outline"
        onClick={[Function]}
        type="button"
      >
        <div
          className="list-menu-item-icon"
        >
          <svg
            height="25"
            width="21"
          >
            <rect
              fill="none"
              height="25"
              stroke="black"
              strokeWidth="3"
              width="21"
              x="0"
              y="0"
            />
            <line
              stroke="green"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="5"
              y2="5"
            />
            <line
              stroke="green"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="10"
              y2="10"
            />
            <line
              stroke="green"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="15"
              y2="15"
            />
            <line
              stroke="green"
              strokeWidth="2"
              x1="3"
              x2="18"
              y1="20"
              y2="20"
            />
          </svg>
        </div>
        <p
          className="list-menu-item-text"
        >
          Low
        </p>
      </button>
    </li>
  </ul>
</div>
`;
